plugins/modules/aws_acm.py

```diff
@@ -362,30 +362,39 @@ def main():
             else:
                 module.debug("Existing certificate in ACM is different, overwriting")
 
-                # update cert in ACM
-                arn = acm.import_certificate(client, module,
+                if module.check_mode:
+                    arn = old_cert['certificate_arn']
+                    # note: returned domain will be the domain of the previous cert
+                else:
+                    # update cert in ACM
+                    arn = acm.import_certificate(client, module,
+                                                 certificate=module.params['certificate'],
+                                                 private_key=module.params['private_key'],
+                                                 certificate_chain=module.params['certificate_chain'],
+                                                 arn=old_cert['certificate_arn'],
+                                                 tags=tags)
+                domain = acm.get_domain_of_cert(client=client, module=module, arn=arn)
+                module.exit_json(certificate=dict(domain_name=domain, arn=arn), changed=True)
+        else:  # len(certificates) == 0
+            module.debug("No certificate in ACM. Creating new one.")
+            if module.check_mode:
+                arn = 'arn:aws:acm:us-east-1:123456789012:certificate/01234567-abcd-abcd-abcd-012345678901'
```

Comment:
Personally I prefer to not return the ARN rather than returning a 'fake' value.

Reply:
What happens if someone uses the ARN normally, and wants to run the playbook in check mode? If the ARN is not returned in check mode, won't that mean that a valid playbook with fail with check_mode on?

Reply:
Yeah, it's a complex question, because if you then try to use a fake ARN somewhere any lookups against that ARN will fail.  Ultimately if you're running in check mode then relying on the output of anything that's supposed to create a resource is going to be unreliable.

Since we don't have a policy around what should be returned in check mode I originally approved the PR, but we're also seeing a test failure.

Reply:
Offering my two cents: Valid playbooks often fail in check mode because a given task will depend on a resource created by a previous one. Since no resources should be created in check mode, some tasks are expected to fail. That's just the nature of the beast. Writers of playbooks already know this, and it's up to them to handle it. The original issue here was that the role was creating resources in check mode, which is a catastrophic flaw by comparison. I agree with @tremble that it's better to return nothing. Returning a fake ARN would likely send the operator on a wild goose chase.

Reply:
Ok. I'll remove that.